plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryCredentialsSupplier.java

```diff
@@ -38,13 +36,10 @@ public BigQueryCredentialsSupplier(Optional<String> credentialsKey, Optional<Str
         requireNonNull(credentialsKey, "credentialsKey is null");
         requireNonNull(credentialsFile, "credentialsFile is null");
         // lazy creation, cache once it's created
-        this.credentialsCreator = Suppliers.memoize(() -> {
-            Optional<Credentials> credentialsFromKey = credentialsKey.map(BigQueryCredentialsSupplier::createCredentialsFromKey);
-            Optional<Credentials> credentialsFromFile = credentialsFile.map(BigQueryCredentialsSupplier::createCredentialsFromFile);
-            return Stream.of(credentialsFromKey, credentialsFromFile)
-                    .flatMap(Streams::stream)
-                    .findFirst();
-        });
+        this.credentialsCreator = Suppliers.memoize(() ->
+            credentialsKey.map(BigQueryCredentialsSupplier::createCredentialsFromKey)
+                    .or(() -> credentialsFile.map(BigQueryCredentialsSupplier::createCredentialsFromFile))
```

Comment:
Do we have test coverage here?

Reply:
I don't think so - the `TestBigQueryIntegrationSmokeTest` creates BQ client without using `BigQueryCredentialsSupplier`: https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BigQueryQueryRunner.java#L120

Reply:
It is a bit risky to refactor code without tests. Would it be possible to you to test it manually?

Reply:
I think this is covered. the BQ client instantiation linked by @sergey-melnychuk is for the testing client we create to directly run queries on BigQuery without using Trino.

The actual client gets created at https://github.com/trinodb/trino/blob/f6422d04663ab011d3ddb831ed16dec02659c47e/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryConnectorModule.java#L66.

Reply:
@hashhar , thanks for pointing out! I was looking at the wrong test :) If this is covered, I allowed myself to have a bit more fun with `Optional`s: db72a4f4be5e2119bdbd937c74733ddc92e97949

After some thinking this would be strange indeed if the only way credentials are provided was not covered with any system-/integration-/e2e- tests.